packages/jsonc/jsonc.test.mjs

```diff
@@ -0,0 +1,110 @@
+import assert from 'node:assert/strict';
+import { spawnSync } from 'node:child_process';
+import { execPath } from 'node:process';
+import { describe, it } from 'node:test';
+import { fileURLToPath } from 'node:url';
+
+// biome-ignore lint: i know what i'm doing 😎
+const stripAnsi = (str) => str.replace(/(\u001b\[[0-9;]*m)/g, '');
+
+const expected = "{ foo: 'bar', baz: 'qux', quux: [ 'corge', 'grault' ] }\n";
+
+describe('jsonc (e2e)', () => {
+	describe('json', () => {
+		const cwd = fileURLToPath(import.meta.resolve('./fixtures'));
+		const e2eTest = fileURLToPath(
+			import.meta.resolve('./fixtures/e2e-json.mjs'),
+		);
+
+		it('should work with `--loader`', () => {
+			const {
+				status: code,
+				stderr,
+				stdout,
+			} = spawnSync(
+				execPath,
+				[
+					'--no-warnings',
+					'--loader',
+					fileURLToPath(import.meta.resolve('./jsonc.mjs')),
+					e2eTest,
+				],
+				{ cwd },
+			);
+
+			assert.equal(stderr.toString(), '');
+			assert.equal(stripAnsi(stdout.toString()), expected);
```

Comment:
You can avoid these toSting()s by setting `encoding: 'utf-8'` in spawnSync's options 🙂

```suggestion
			assert.equal(stdout, expected);
```